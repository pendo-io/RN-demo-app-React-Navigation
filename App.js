import * as React from 'react';
import Login from './components/Login';
import Plan from './components/Plan';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {PendoSDK, NavigationLibraryType, WithPendoReactNavigation} from 'pendo';
import {themes} from "./components/ThemeContext";
import {ThemeProvider} from "./components/ThemeContext";

function initPendo() {
    PendoSDK.setDebugMode(true);
    const navigationOptions = {library: NavigationLibraryType.ReactNavigation};
    const pendoKey = 'eyJhbGciOiJSUzI1NiIsImtpZCI6IiIsInR5cCI6IkpXVCJ9.eyJkYXRhY2VudGVyIjoidXMiLCJrZXkiOiJmZWRiN2YyMDY1Y2FmYWFjZGFlZWNhMDNmOGRkNzgyNzRiNWEwNzMzZmIyMWUzNTgwOGViY2Y0YThiNTg3MjVjNmNkNzdmYzYyMDMyNmIyYjg0ZWU2YjNiNmZhY2I5OGNkZWRmMmJlOTRlMzFlNGNlNjNjYTE4NWZkNTIyYzJmM2E0OWIwMzA3YzkwZTA0OTgxY2Q1NDQ4ZWU5ZGIzYjA1YTg2OTQzYzgwYzBjZDA0YTAwNzU2MjJlMjRiYjFmMDViOGU3MzlhY2QwM2U3NmIwZjUxMGExMzBjNzI4NTVhNi5hYmMwYzk4OWVkMGFjOTAxNzNiNTMwYmEzOGIxZWJhNy41MzU1NzQwOWQ2ZjdlMTM4MjczNDkxZDc2NjhjYTNmNzJmZmJkYjU5OGMwYWJiMDk0ZTFlNDg0YzVlMWIwNGM3In0.em6YOaXKpEDAwmOCdf4CTIRDpyKbs9NNkVIZCDN6UwCkXgwtToaLO2PUoEmG0FG7te8d-i1qoowIKidCecuIb0hS9fooKHhyV36RdNGF-Icr4zs1Iw9jU7_asgZybajs7FKl7klJjmTzER8u3wRPVxiLopJG0SWCym_qw1KTaQ4';
    PendoSDK.setup(pendoKey, navigationOptions);
}

initPendo();
const Stack = createNativeStackNavigator();

function App() {
    const PendoNavigationContainer = WithPendoReactNavigation(NavigationContainer);
    return (
        <ThemeProvider>
            <PendoNavigationContainer theme={themes.dark}>
                <Stack.Navigator
                    initialRouteName="Login"
                    screenOptions={{
                        headerShown: false,
                    }}>
                    <Stack.Screen
                        name="Login"
                        component={Login}
                    />
                    <Stack.Screen
                        name='Plan'
                        component={Plan}/>
                </Stack.Navigator>
            </PendoNavigationContainer>
        </ThemeProvider>
    );
}

export default App;
