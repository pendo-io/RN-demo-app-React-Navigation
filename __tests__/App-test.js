/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
//import {PendoSDK, NavigationLibraryType} from 'rn-pendo-sdk';
import {create, act} from 'react-test-renderer';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {idObj as PendoSDK} from 'identity-obj-proxy';


it('renders correctly', async () => {
  let tree
  act(()=>{
     tree = renderer.create(<App />);
  })
  
  expect(tree.toJSON()).toMatchSnapshot();
});
