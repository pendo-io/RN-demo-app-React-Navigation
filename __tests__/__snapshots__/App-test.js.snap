// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#2f3641",
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ec2059",
                "fontSize": 25,
                "fontWeight": "bold",
                "padding": 40,
              }
            }
          >
            Login
          </Text>
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Username"
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "margin": 12,
                  "padding": 10,
                }
              }
              value=""
            />
            <TextInput
              onChangeText={[Function]}
              placeholder="Password"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "margin": 12,
                  "padding": 10,
                }
              }
              value=""
            />
            <View
              style={
                Object {
                  "paddingHorizontal": 135,
                  "paddingTop": 25,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "margin": 8,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#ec2059",
                        },
                      ]
                    }
                  >
                    login
                  </Text>
                </View>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
